sdk/importer.go

```diff
@@ -33,8 +33,9 @@ func (c *ImportCandidate) Equal(other ImportCandidate) bool {
 }
 
 type ImportSource struct {
-	Env   []string
-	Files []string
+	Env      []string
+	Files    []string
+	AWSVault bool
```

Comment:
I think we need to figure out a better general way of dealing with import sources that are neither files or environment variables. In this case, we are dealing with importing credentials from password managers. My take is that we should add another option in `ImportSource` called `PasswordManagers []string` which would be a list of names of password managers that contain our desired credentials. 

The goal of this would be for the user to see the following when they are importing from password managers/vaults:

```
> Password Manager ( profile andy)  AWS-Vault - MacOS Keychain
   Password Manager ( profile dev)    AWS-Vault - MacOS Keychain
   File                                                      ~/.aws/credentials
```

However, since this is an SDK change that will require a small change in `op` as well, I'm fine with decoupling this from the current PR. For this PR, I would then be ok with the outcome:

```
> File (profile andy)   AWS-Vault - MacOS Keychain
   File (profile dev)      AWS-Vault - MacOS Keychain
```
Even though this is not nice, you could argue that the keychain works with encrypted files under the hood, so we can keep this until we introduce a more suitable import source.

What do you think @florisvdg ?

Reply:
As for import sources in this SDK, we can either introduce "Other" and have the plugin author give a name of the source (e.g. "macOS Keychain") to present in `op`. But we could introduce new sources for "macOS Keychain", "GNOME Keyring", and "KWallet" in the SDK. May be the nicer option.

The source value should indeed probably be/contain the profile name. But not sure if "AWS Vault" should even be mentioned at all here.

Reply:
For now, I made it so that a new `Other` import source is introduced in the SDK and used in the AWS Vault importer.

When a user attempts to import for AWS, the source value for the items retrieved from the vaulting backend are displayed as `<vaulting backend> (profile)`, with no second column. Again, this requires some updates in the 1Password CLI codebase as well:

![image](https://user-images.githubusercontent.com/53918631/229223359-79f0987f-62e3-4276-93a4-c74a41230218.png)

Let me know your thoughts @AndyTitu @florisvdg!

Reply:
Makes sense to me 👍 

One thing about how it's displayed in `op`: it would be nice if it would be spaced just like `File` and `Environment`.
```
macOS Keychain         work
macOS Keychain         home
File                   ~/.aws/credentials
```
Could be worth introducing something like:
```go
type CustomSource struct {
   Type  string
   Value []string
}
```
Or alternatively (maybe simpler), populate the `Other []string` value as: `[]string{"macOS Keychain", "work"}`.

Reply:
> One thing about how it's displayed in op: it would be nice if it would be spaced just like File and Environment.

Yes, let's do this.

> Could be worth introducing something like:
```
type CustomSource struct {
   Type  string
   Value []string
}
```

I like this clear definition the best

@williamhpark

Reply:
The reason why I thought it would be better to format in just one column in `op` was to match what is currently done for the file importer, where the AWS profile name is displayed in parentheses next to the source type (`File (test)` in the screenshot):

![image](https://user-images.githubusercontent.com/53918631/229572551-8ce290b4-ed1b-4af7-a0c1-c8b54474fd7f.png)

This being said, I think that introducing the `CustomSource` that @florisvdg suggested would be useful for potential future `Other` sources anyways. But in this case, I feel inclined to keep it as one column by passing a blank string as the value, i.e.:

`attempt := out.NewAttempt(importer.SourceOther(keyringBackend, ""))`

Reply:
I pushed a change based on my last comment, let me know if this seems reasonable @AndyTitu @florisvdg.

Reply:
Ah gotcha. Okay to keep it consistent with `File` then.

I do think we need to revisit this approach in the future. Ideally the profile name for both of them would be in the column on the right, instead of in parentheses on the left.

Reply:
Hmm, I see.

I am still inclined to think specifying the value of the import source with the profile name would be slightly more suitable. It doesn't make a huge difference, I'll admit, but it seems just a bit more consistent.

Here is my reasoning based on looking at how a keychain access is organised
<img width="727" alt="Screenshot 2023-04-04 at 20 29 58" src="https://user-images.githubusercontent.com/45081667/229877713-f00c29d0-0e4a-4081-9aa1-5d15b2aa214a.png">

1. The name of the keychain is `aws-vault (andy)`. If we strip `aws-vault` as we talked about we remain with just andy. In the case of a `File` importer `.aws/credentials` is also the name of the file. So it seems more consistent to treat the names (i.e "andy" and other profile names) as import source values, like we do with file names

2. We can notice a third element in the keychain which is `Sts.GetSessionToken` (temporary credentials). In this case we don't want to import those as well, but if we did would it make sense to add that as a name hint rather than as a name? That's not a profile anymore.

Reply:
@AndyTitu I see what you mean for point 1. When stored in Pass through AWS Vault, the credentials are stored under the profile name as well (`pass-profile` in the ss) , which adds to your point that the profile name should be treated as the import source value:

![image](https://user-images.githubusercontent.com/53918631/229898568-facbc920-82e3-4ac8-9afd-c2fce1d313fb.png)

To keep consistency with the file importer, I'm still inclined to keep the profile name as the name hint. Although it seems a bit redundant, what are your thoughts on having the profile name as both the name hint and source value? Like so:

![image](https://user-images.githubusercontent.com/53918631/229899137-4711b82f-98e8-42b9-8852-ac723e43edfc.png)

Reply:
Or another option is we could format like so, similar to what @AndyTitu suggested at the beginning of this thread:

```
> Password Manager (work)       macOS Keychain
  Password Manager (home)       Pass
  File                          ~/.aws/credentials
```

I think I like this best actually.

Reply:
I don't think seeing `Password Manager` there is that helpful to the user. I'd suggest to stick with your current approach until we can think of something nicer in the future:

![](https://user-images.githubusercontent.com/53918631/229223359-79f0987f-62e3-4276-93a4-c74a41230218.png)

Reply:
I'd say just the name hint. I don't see the value of duplicating it.

Reply:
👍